rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -679,6 +712,46 @@ impl FirmwareProcessor {
         Ok(*manifest)
     }
 
+    /// Load the manifest from MCU SRAM using DMA
+    ///
+    /// # Returns
+    ///
+    /// * `Manifest` - Caliptra Image Bundle Manifest
+    #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
+    fn load_manifest_from_mcu(
+        persistent_data: &mut PersistentDataAccessor,
+        soc_ifc: &mut SocIfc,
+        dma: &mut Dma,
+    ) -> CaliptraResult<ImageManifest> {
+        let manifest = &mut persistent_data.get_mut().manifest1;
+        let manifest_buf = manifest.as_mut_bytes();
+
+        // Get MCU SRAM address
+        let mci_base_addr: AxiAddr = soc_ifc.mci_base_addr().into();
+        let recovery_interface_base_addr: AxiAddr = soc_ifc.recovery_interface_base_addr().into();
+        let caliptra_base_addr: AxiAddr = soc_ifc.caliptra_base_axi_addr().into();
+
+        // Read manifest from MCU SRAM using DMA directly into manifest buffer
+        let manifest_size_words = manifest_buf.len().div_ceil(4);
+        let manifest_words = unsafe {
+            core::slice::from_raw_parts_mut(
+                manifest_buf.as_mut_ptr() as *mut u32,
+                manifest_size_words,
+            )
+        };
```

Comment:
You should be able to avoid the unsafe here using zerocopy:
```suggestion
        let (manifest_words, remainder) = <[u32]>::mut_from_prefix(&manifest_buf);
```